iib/web/api_v1.py

```diff
@@ -624,3 +626,48 @@ def add_rm_batch():
         ', '.join(processed_request_ids),
     )
     return flask.jsonify(request_jsons), 201
+
+
+@api_v1.route('/builds/merge-index-image', methods=['POST'])
+@login_required
+def merge_index_image():
+    """
+    Submit a request to merge two index images.
+
+    :rtype: flask.Response
+    :raise ValidationError: if required parameters are not supplied
+    """
+    payload = flask.request.get_json()
+    if not isinstance(payload, dict):
+        raise ValidationError('The input data must be a JSON object')
+
+    request = RequestMergeIndexImage.from_json(payload)
+    db.session.add(request)
+    db.session.commit()
+    messaging.send_message_for_state_change(request, new_batch_msg=True)
+
+    overwrite_from_index = _should_force_overwrite() or payload.get('overwrite_from_index')
+    celery_queue = _get_user_queue(serial=overwrite_from_index)
+    args = [
+        payload['binary_image'],
+        payload['source_from_index'],
+        payload['target_index'],
+        payload.get('deprecation_list'),
+        request.id,
+        overwrite_from_index,
+        payload.get('overwrite_from_index_token'),
+    ]
+    safe_args = copy.copy(args)
```

Comment:
Why not simply call _get_safe_args here?

Reply:
Fixed